pkg/migrations/migrations.go

```diff
@@ -46,6 +46,16 @@ type RequiresSchemaRefreshOperation interface {
 	RequiresSchemaRefresh()
 }
 
+type SQLTransformer interface {
+	Transform(sql string) (string, error)
+}
+
+type SQLTransformerFunc func(string) (string, error)
+
+func (fn SQLTransformerFunc) Transform(sql string) (string, error) {
```

Comment:
nit (feel free to ignore): `TransformSQL` may be a better name to avoid confusion